packages/nimble-components/src/table/template.ts

```diff
@@ -39,97 +39,112 @@ export const template = html<Table>`
-                                    record-id="${(x, c) => c.parent.tableData[x.index]?.id}"
-                                    ?selectable="${(_, c) => c.parent.selectionMode !== TableRowSelectionMode.none}"
-                                    ?selected="${(x, c) => c.parent.tableData[x.index]?.selectionState === TableRowSelectionState.selected}"
-                                    ?hide-selection="${(_, c) => c.parent.selectionMode !== TableRowSelectionMode.multiple}"
-                                    :dataRecord="${(x, c) => c.parent.tableData[x.index]?.record}"
-                                    :columns="${(_, c) => c.parent.columns}"
-                                    :nestingLevel="${(x, c) => c.parent.tableData[x.index]?.nestingLevel}"
-                                    @click="${(x, c) => c.parent.onRowClick(x.index, c.event as MouseEvent)}"
-                                    @row-selection-toggle="${(x, c) => c.parent.onRowSelectionToggle(x.index, c.event as CustomEvent<TableRowSelectionToggleEventDetail>)}"
-                                    @row-action-menu-beforetoggle="${(x, c) => c.parent.onRowActionMenuBeforeToggle(x.index, c.event as CustomEvent<TableActionMenuToggleEventDetail>)}"
-                                    @row-action-menu-toggle="${(_, c) => c.parent.onRowActionMenuToggle(c.event as CustomEvent<TableActionMenuToggleEventDetail>)}"
+                            <span role="gridcell">
+                                <${buttonTag}
+                                    class="collapse-all-button ${x => `${x.showCollapseAll ? 'visible' : ''}`}"
+                                    content-hidden
+                                    appearance="${ButtonAppearance.ghost}"
+                                    title="${x => tableGroupsCollapseAllLabel.getValueFor(x)}"
+                                    @click="${x => x.handleCollapseAllGroupRows()}"
                                 >
-                                ${when((x, c) => (c.parent as Table).openActionMenuRecordId === (c.parent as Table).tableData[x.index]?.id, html<VirtualItem, Table>`
-                                    ${repeat((_, c) => (c.parent as Table).actionMenuSlots, html<string, Table>`
-                                        <slot
-                                            name="${x => x}"
-                                            slot="${x => `row-action-menu-${x}`}">
-                                        </slot>
+                                    <${iconTriangleTwoLinesHorizontalTag} slot="start"></${iconTriangleTwoLinesHorizontalTag}>
+                                    ${x => tableGroupsCollapseAllLabel.getValueFor(x)}
+                                </${buttonTag}>
+                            </span>
+                        </span>
+                        <span class="column-headers-container" ${ref('columnHeadersContainer')}>
+                            ${repeat(x => x.visibleColumns, html<TableColumn>`
```

Comment:
think you can specify the parent type here and avoid some `as` casts
```suggestion
                            ${repeat(x => x.visibleColumns, html<TableColumn, Table>`
```